src/main/java/com/github/hanleyt/JerseyExtension.java

```diff
@@ -1,80 +1,103 @@
-        return context.getStore(ExtensionContext.Namespace.GLOBAL);
-    }
-
-}
+package com.github.hanleyt;
+
+import org.glassfish.jersey.client.ClientConfig;
+import org.glassfish.jersey.test.JerseyTest;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolver;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Application;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public class JerseyExtension implements BeforeEachCallback, AfterEachCallback, ParameterResolver {
+
+    private static final Collection<Class<?>> INJECTABLE_PARAMETER_TYPES = Arrays.asList(Client.class, WebTarget.class, URI.class);
+
+    private final Function<ExtensionContext, Application> applicationProvider;
+    private final BiFunction<ExtensionContext, ClientConfig, ClientConfig> configProvider;
+
+    private JerseyExtension() {
+        throw new IllegalStateException("JerseyExtension must be registered programmatically");
+    }
+
+    public JerseyExtension(Supplier<Application> applicationSupplier) {
+        this.applicationProvider = (unused) -> applicationSupplier.get();
+        this.configProvider = null;
+    }
+
+    public JerseyExtension(Supplier<Application> applicationSupplier, BiFunction<ExtensionContext, ClientConfig, ClientConfig> configProvider) {
+        this.applicationProvider = (unused) -> applicationSupplier.get();
+        this.configProvider = configProvider;
+    }
+
+    public JerseyExtension(Function<ExtensionContext, Application> applicationProvider) {
+        this.applicationProvider = applicationProvider;
+        this.configProvider = null;
+    }
+
+    public JerseyExtension(Function<ExtensionContext, Application> applicationProvider, BiFunction<ExtensionContext, ClientConfig, ClientConfig> configProvider) {
+        this.applicationProvider = applicationProvider;
+        this.configProvider = configProvider;
+    }
+
+    @Override
+    public void beforeEach(ExtensionContext context) throws Exception {
+        JerseyTest jerseyTest = initJerseyTest(context);
+        getStore(context).put(Client.class, jerseyTest.client());
+        getStore(context).put(WebTarget.class, jerseyTest.target());
+    }
+
+    private JerseyTest initJerseyTest(ExtensionContext context) throws Exception {
+        JerseyTest jerseyTest = new JerseyTest() {
+            @Override
+            protected Application configure() {
+                getStore(context).put(URI.class, getBaseUri());
+                return applicationProvider.apply(context);
+            }
+
+			@Override

```

Comment:
indentation seems a bit off here?

Reply:
I was using tabs in indentation (Eclipse's default setting), so that was throwing the indentation off.  I've converted them to spaces and committed the change.  Sorry about that.